src/Spd.Resource.Repository/PortalUser/PortalUserRepository.cs

```diff
@@ -95,7 +91,25 @@ private async Task<PortalUserResp> UpdatePortalUserAsync(UpdatePortalUserCmd c,
             identity = await _context.GetIdentityById((Guid)c.IdentityId, ct);
             _context.SetLink(portalUser, nameof(portalUser.spd_IdentityId), identity);
         }
+
+        ContactRoleCode? currentContactRoleCode = SharedMappingFuncs.GetContactRoleCode(portalUser.spd_spd_role_spd_portaluser);

```

Comment:
why we need GetContactRoleCode func, why put into Shared?
you just needs to check if role changed, use DynamicsContextLookupHelpers role part should be enough.

Reply:
I use that function because it is a cleaner way to get the role assigned to the portal user (currently, the relationship is defined as a collection, even though it is a 1 to 1 relationship). I placed the function into shared, because it is also used in the portal user mapping. If you still believe it is not a good idea, I can apply your suggestion.